packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -816,25 +886,28 @@ export class PaginatedMessage {
 	): Promise<void> {
 		if (interaction.user.id === targetUser.id) {
 			const action = this.actions.get(interaction.customId)!;
-			const previousIndex = this.index;
-
-			await action.run({
-				interaction,
-				handler: this,
-				author: targetUser,
-				channel,
-				response: this.response!,
-				collector: this.collector!
-			});
 
-			const newIndex = previousIndex === this.index ? previousIndex : this.index;
-			const messagePage = await this.resolvePage(newIndex);
-			const updateOptions = isFunction(messagePage) ? await messagePage(newIndex, this.pages, this) : messagePage;
+			if (actionIsButtonOrMenu(action)) {
+				const previousIndex = this.index;
 
-			if (interaction.replied || interaction.deferred) {
-				await interaction.editReply(updateOptions);
-			} else {
-				await interaction.update(updateOptions);
+				await action.run({
+					interaction,
+					handler: this,
+					author: targetUser,
+					channel,
+					response: this.response!,
+					collector: this.collector!
+				});
+
+				const newIndex = previousIndex === this.index ? previousIndex : this.index;
```

Comment:
Doesn't this end up as this.index in both cases?

Reply:
uuhhhh.. that's a good point.